ml-proto/spec/check.ml

```diff
@@ -134,9 +134,9 @@ let rec check_expr c et e =
     check_expr_opt c (label c x) eo e.at
 
   | Br_if (x, eo, e) ->
+    check_expr_opt c et eo e.at;
```

Comment:
@lukewagner Reverting https://github.com/WebAssembly/spec/pull/180 is orthogonal to your pre/post order concerns. To keep the AST clean for both pre/post order it is simply necessary that the encoding of each operator stands on its own and does not depend on context either way - I would support a PR to make this a design principle. For example, we can revert https://github.com/WebAssembly/spec/pull/180 and use the v8 encoding for breaks and fill with the nop, and this works pre/post order. It was only the high interest in the explicit encoding efficiency at the AST level that demanded br0.

Expression results need to be one entry on the expression stack, otherwise you have a stack machine!  This is needed to be able to discard them as a set and to consume only the first-n, and to pass them through the 'parameterized' operators. For example: to use `if_else` and `select` to choose between sets of values, and to be able to pass a set from a block, and with a good `br_if` design to be able to direct a set of values.

@sunfishcode It affects usability of the AST if the consumers of `br_if` values can not independently decide to discard values - as you seem to have already experienced.